core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -9479,14 +9479,13 @@ protected List<User> doGetUserListOfRoleWithID(String roleName, String filter) t
     }
 
     /**
-     * Return the count of users belong to the given role for the given filter.
+     * Return the count of users belong to the given role for the given filter when unique id feature is not enabled.
      *
      * @param roleName role name.
-     * @param filter   filter.
      * @return user count for the given role.
      * @throws UserStoreException Thrown by the underlying UserStoreManager.
      */
-    protected int doGetUserCountOfRole(String roleName, String filter) throws UserStoreException {
+    protected int doGetUserCountOfRole(String roleName) throws UserStoreException {
```

Comment:
Lets not change the signature of protected/public methods. Instead We can overload the method

Reply:
- This method was introduced by the PR https://github.com/wso2/carbon-kernel/pull/3906 recently and this method is to be utilized by only `getUserCountForRole` method which will be called by the SCIM2 repo with the changes introduced in the PR https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/533.